src/GitVersionCore/Git/ReferenceName.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+
+namespace GitVersion
+{
+    public class ReferenceName
+    {
+        private const string LocalBranchPrefix = "refs/heads/";
+        private const string RemoteTrackingBranchPrefix = "refs/remotes/";
+        private const string TagPrefix = "refs/tags/";
+
+        public ReferenceName(string canonicalName)
+        {
+            CanonicalName = canonicalName;
+            FriendlyName = Shorten();
+            NameWithoutRemote = RemoveRemote();
+        }
+        public string CanonicalName { get; }
+        public string FriendlyName { get; }
+        public string NameWithoutRemote { get; }
```

Comment:
Perhaps we can remove `Name` in these properties as they will always be resolved with `Name.` as a prefix?

```suggestion
        public string Canonical { get; }
        public string Friendly { get; }
        public string WithoutRemote { get; }
```

Reply:
Agree with that

Reply:
Done